.github/workflows/ci_cd.yml

```diff
@@ -103,14 +105,20 @@ jobs:
               if "ingestion_server" in changes:
                   publish_matrix["image"].append("ingestion_server")
 
+          do_build = 'true' if len(build_matrix["image"]) != 0 else 'false'
           build_matrix = json.dumps(build_matrix)
+          do_publish = 'true' if len(publish_matrix["image"]) != 0 else 'false'
           publish_matrix = json.dumps(publish_matrix)
 
+          print(f"do-build={do_build}")
           print(f"build-matrix={build_matrix}")
+          print(f"do-publish={do_publish}")
           print(f"publish-matrix={publish_matrix}")
 
           with open(os.environ.get("GITHUB_OUTPUT"), "a") as gh_out:
+              print(f"do-build={do_build}", file=gh_out)
```

Comment:
Just curious, why do we use `print` and not `write` here?

Reply:
I didn't think much about it, I mapped what one would usually do in Bash (`echo`) to the Python equivalent. I suppose `write` would've worked just as well.

Reply:
I _think_ you wouldn't need to pass the file handler, too.